### PR TITLE
feat(upload): create upload modal

### PR DESCRIPTION
**Pull request by luqven**
## Description

This PR creates the upload modal in the `<Dialog />` component. This modal is displayed whenever the user clicks on `Upload` in the header. The modal allows the user to

1. Change the destination source
2. Change the destination path
3. Preview the asset before upload

This PR also adds `uploadForm` to state.

```typescript
  uploadForm: {
    file?: any;
    source?: {
      id?: string;
      name?: string;
      domain?: string;
    };
    previewSource?: string;
  };
```

This PR also adds some helper class methods to manage state.

```js
setIsUploading = (value: boolean) => {
    this.setState({ isUploading: value });
  };

  setUploadSource = (source: { id: string; name: string; domain: string }) => {
    const uploadForm = { ...this.state.uploadForm, source };
    this.setState({ uploadForm });
  };

  updateFileForm = (file: any, previewSource: string, isUploading: boolean) => {
    const uploadForm = {
      ...this.state.uploadForm,
      file,
      source: this.state.selectedSource,
      previewSource,
    };
    this.setState({ uploadForm, isUploading });
  };
```

A future PR should add a callback on upload confirmation that actually sends the file to the imgix asset manager via then management SDK.

# Before
No upload modal.

# After
The upload modal is displayed when the user clicks upload

# Steps to test
- pull down the branch
- run `npm run start`
- open contentful staging env
- click "add image"
- click "upload"
- see modal open

# 🖼️  Screenshots

https://user-images.githubusercontent.com/16711614/198851677-a5c5b739-6af3-4218-ad4d-a47bf856d62f.mov

## Checklist

- [x] Read the [contributing guidelines](CONTRIBUTING.md).
- [x] Each commit follows the [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) spec format.
- ~[ ] Update the readme (if applicable).~
- ~[ ] Update or add any necessary API documentation (if applicable)~
- [x] All existing unit tests are still passing (if applicable).

- [ ] Add some [steps](#steps-to-test) so we can test your bug fix or feature (if applicable).
- [ ] Add new passing unit tests to cover the code introduced by your PR (if applicable).
- ~[ ] Any breaking changes are specified on the commit on which they are introduced with `BREAKING CHANGE` in the body of the commit.~
- ~[ ] If this is a big feature with breaking changes, consider opening an issue to discuss first. This is completely up to you, but please keep in mind that your PR might not be accepted.~

### CODE REVIEW

**Comment by luqven** [comment]
Current dependencies on/for this PR:
* next
  * **PR #140** <a href="https://app.graphite.dev/github/pr/imgix/contentful/140" target="_blank">Graphite</a> 
    * **PR #139** <a href="https://app.graphite.dev/github/pr/imgix/contentful/139" target="_blank">Graphite</a> 
      * **PR #141** <a href="https://app.graphite.dev/github/pr/imgix/contentful/141" target="_blank">Graphite</a> 
        * **PR #142** <a href="https://app.graphite.dev/github/pr/imgix/contentful/142" target="_blank">Graphite</a> 
          * **PR #143** <a href="https://app.graphite.dev/github/pr/imgix/contentful/143" target="_blank">Graphite</a> 
            * **PR #144** <a href="https://app.graphite.dev/github/pr/imgix/contentful/144" target="_blank">Graphite</a> 
              * **PR #145** <a href="https://app.graphite.dev/github/pr/imgix/contentful/145" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/imgix/contentful/145?utm_source=stack-comment).

**Comment by luqven** [comment]
Current dependencies on/for this PR:
* next
  * **PR #140** <a href="https://app.graphite.dev/github/pr/imgix/contentful/140" target="_blank">Graphite</a> 
    * **PR #139** <a href="https://app.graphite.dev/github/pr/imgix/contentful/139" target="_blank">Graphite</a> 
      * **PR #141** <a href="https://app.graphite.dev/github/pr/imgix/contentful/141" target="_blank">Graphite</a> 
        * **PR #142** <a href="https://app.graphite.dev/github/pr/imgix/contentful/142" target="_blank">Graphite</a> 
          * **PR #143** <a href="https://app.graphite.dev/github/pr/imgix/contentful/143" target="_blank">Graphite</a> 
            * **PR #144** <a href="https://app.graphite.dev/github/pr/imgix/contentful/144" target="_blank">Graphite</a> 
              * **PR #145** <a href="https://app.graphite.dev/github/pr/imgix/contentful/145" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/imgix/contentful/145?utm_source=stack-comment).

File: src/components/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -405,6 +431,60 @@ export default class Dialog extends Component<DialogProps, DialogState> {
...
+            <div className="ix-upload-editor">
+              <div className="ix-upload-title-bar">
+                <p className="ix-upload-header-text">Upload to Asset Manager</p>
+                <Button
+                  size="small"
+                  buttonType="naked"
+                  icon="Close"
+                  className="ix-close-button"
+                  onClick={() => this.setIsUploading(false)}
+                ></Button>
```
Comment (sherwinski) [inline src/components/Dialog/Dialog.tsx:445]: Can we conditionally disable this button if `isUploading` is true?

**Review by sherwinski — APPROVED** [review]

**Comment by luqven** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/imgix/contentful/145?utm_source=stack-comment).


Job Reference: yGgqIySrcaAmVh8uxOCg

**Pull request merged**